spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/support/SimpleSpannerRepository.java

```diff
@@ -105,7 +113,39 @@ public SpannerOperations getSpannerTemplate() {
 	@Override
 	public boolean existsById(ID key) {
 		Assert.notNull(key, "A non-null ID is required.");
-		return findById(key).isPresent();
+		return doIfKey(key, (k) -> {
+
+			final SpannerPersistentEntity<?> persistentEntity = this.spannerTemplate.getMappingContext()
+					.getPersistentEntity(this.entityType);
+
+			int tagNum = 0;
+			List<String> columns = new ArrayList<>();
+			List<String> tags = new ArrayList<>();
+			List<Object> keyParts = new ArrayList<>();
+			List<SpannerPersistentProperty> properties = persistentEntity.getFlattenedPrimaryKeyProperties();
+			Iterator parentKeyParts = k.getParts().iterator();
+			while (parentKeyParts.hasNext()) {
```

Comment:
This logic overlaps with `SpannerStatementQueryExecutor.getChildrenRowsQuery` significantly. Please refactor to avoid duplication. 
Also, I think the bulk of this method should be moved to `SpannerTemplate`. Feel free to introduce `SpannerTemplate.existsById` method.

Reply:
Today I think the better idea is to move the "existsById" logic to the "SpannerStatementQueryExecutor"
It is very similar to the method "SpannerStatementQueryExecutor#getChildrenRowsQuery" (that actually should be better named like "getChildrenRowsStatement")

I'll do another PR tomorrow I believe.